docs/reST/ref/display.rst

```diff
@@ -573,47 +573,48 @@ required).
 
-   | :sl:`Return whether SDL is hinted to enable or disable the screensaver.`
+   | :sl:`Return whether pygame is hinted to enable or disable the screensaver.`
    | :sg:`get_screensaver_enabled() -> bool`
 
-   Return whether SDL is hinted to enable or disable the screensaver.
-   This function is only available with SDL2.  Under SDL1, this
-   function will raise an exception.
+   Return whether screensaver is enabled. The default is False.
+   By default pygame tries to turn off the screensaver during game play.
 
-   .. note:: The reliability of this function is subject to platform
-             support.  Please see
+   .. note:: Some platforms do not have a screensaver or support
+             disabling the screensaver.  Please see
              :func:`pygame.display.set_screensaver_enabled()` for
              caveats with screensaver support.
 
-   .. versionadded:: 2.0
+   .. versionadded:: 2.0 This function is only available with SDL2. Under SDL1,
+   this function will raise a NotImplementedError exception.
 
    .. ## pygame.display.get_screensaver_enabled ##
 
 .. function:: set_screensaver_enabled
 
    | :sl:`Change request state for whether screensaver may run`
-   | :sg:`set_screensaver_enabled() -> None`
+   | :sg:`set_screensaver_enabled(bool) -> None`
 
-   Change the SDL hint for whether screensavers should be enabled or
-   disabled.  This function is only available with SDL2.  Under SDL1,
-   this function will raise an exception.
+   Change whether screensavers should be enabled or disabled. The default is False.
+   By default pygame tries to turn off the screensaver during game play.
 
-   If the screen has been disabled due to this function (or by default
-   in SDL), it will automatically be re-enabled when
-   :func:``pygame.quit()` is called.
+   If the screen has been disabled due to this function, it will automatically
+   be re-enabled when :func:``pygame.quit()` is called.
 
-   It is possible to influence the default value under SDL 2.0.2 or
-   later via the environment variable ``SDL_HINT_VIDEO_ALLOW_SCREENSAVER``,
-   which can be set to either ``0`` (disable) or ``1`` (enable).
+   It is possible to influence the default value via the environment variable
+   ``SDL_HINT_VIDEO_ALLOW_SCREENSAVER``, which can be set to either ``0`` (disable)
+   or ``1`` (enable).
 
    .. note:: Disabling screensaver is subject to platform support.
              When platform support is absent, this function will
              silently appear to work even though the screensaver state
              is unchanged.  The lack of feedback is due to SDL not
              providing any supported method for determining whether
              it supports changing the screensaver state.
+             ``SDL_HINT_VIDEO_ALLOW_SCREENSAVER`` is available in SDL 2.0.2 or later.
+
+   .. versionadded:: 2.0 This function is only available with SDL2. Under SDL1,
+   this function will raise a NotImplementedError exception.
```

Comment:
The code currently uses `TypeError`.  As I understand it, it is the correct error in this case as the function will never be implemented for SDLv1 https://docs.python.org/3/library/exceptions.html#TypeError


> exception TypeError
> 
>     Raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.
> 
>     This exception may be raised by user code to indicate that an attempted operation on an object is not supported, and is not meant to be. If an object is meant to support a given operation but has not yet provided an implementation, NotImplementedError is the proper exception to raise.
> 
>     Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError, but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a ValueError.
> 


vs.



> exception NotImplementedError
> 
>     This exception is derived from RuntimeError. In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.
> 
>     Note
> 
>     It should not be used to indicate that an operator or method is not meant to be supported at all – in that case either leave the operator / method undefined or, if a subclass, set it to None.
> 
>     Note
> 
>     NotImplementedError and NotImplemented are not interchangeable, even though they have similar names and purposes. See NotImplemented for details on when to use it.

Reply:
I guess since it's a hint, we can just have the function do nothing under SDL1? This would make it easier to use from user code that wanted to support pygame 2 and 1.9.x.

Reply:
That is certainly also an option. :)  For the "getter", should we then emulate SDL2 behaviour or just always return a fixed boolean value?

Reply:
Yeah, cool. I think emulating SDL2 behaviour is a good idea.